advanced/Scripts/update.sh

```diff
@@ -144,27 +144,29 @@ main() {
     local binary
     binary="pihole-FTL${funcOutput##*pihole-FTL}" #binary name will be the last line of the output of get_binary_name (it always begins with pihole-FTL)
 
-    if FTLcheckUpdate "${binary}"; then
-        FTL_update=true
-        echo -e "  ${INFO} FTL:\\t\\t${COL_YELLOW}update available${COL_NC}"
-    else
-        case $? in
-            1)
-                echo -e "  ${INFO} FTL:\\t\\t${COL_LIGHT_GREEN}up to date${COL_NC}"
-                ;;
-            2)
-                echo -e "  ${INFO} FTL:\\t\\t${COL_LIGHT_RED}Branch is not available.${COL_NC}\\n\\t\\t\\tUse ${COL_LIGHT_GREEN}pihole checkout ftl [branchname]${COL_NC} to switch to a valid branch."
-                ;;
-            3)
-                echo -e "  ${INFO} FTL:\\t\\t${COL_LIGHT_RED}Something has gone wrong, cannot reach download server${COL_NC}"
-                exit 1
-                ;;
-            *)
-                echo -e "  ${INFO} FTL:\\t\\t${COL_LIGHT_RED}Something has gone wrong, contact support${COL_NC}"
-                exit 1
-        esac
-        FTL_update=false
-    fi
+    local ftl_needs_update
+    ftl_needs_update=$(FTLcheckUpdate "${binary}") #check if FTL needs an update
+    case "${ftl_needs_update}" in
```

Comment:
How does this work? `ftl_needs_update` will contain the returned `output` of the function and the `case` uses this. However, the function has been designed to return these numbers as *exit codes*, so something like `$?` is required here.

Reply:
You're correct. I pushed an update.